reports/spec2.tex

```diff
@@ -148,6 +148,7 @@ \subsection{Checking the Specification}
 forest. The graphs in Figures~\ref{fig:tricky1}--\ref{fig:tricky2} do not
 represent chains.  The graph~\texttt{I1} is a direct-acyclic graph but not a
 forest. The graph~\texttt{I2} has a loop.
+\rs{Perhaps this was something that we failed to make clear, but there is no need to consider graphs that are not trees. Or perhaps I don't fully understand the point that we are making above. Let's discuss.}
```

Comment:
I can see how you would be able to make a side-argument that considering only trees is enough, but I don't think it's an assumption that the Python spec makes.

Otherwise, e.g., `is_complete_chain()` would be vacuously true?

Reply:
@thpani
Could you please check the footnote I added above explaining why we considered forests as well.

Reply:
Looks good, thanks!